### DISCUSSION TITLE
dotnet list package / NuGet.CommandLine.XPlat build

### DISCUSSION THREAD

**Discussion by paulomorgado** [NuGet/Home#12903] [Q&A]
[dotnet list package](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-list-package) / [NuGet.CommandLine.XPlat](https://github.com/NuGet/NuGet.Client/tree/dev/src/NuGet.Core/NuGet.CommandLine.XPlat), apparently, builds the project using .NET SDK.

For projects that have dependencies on tools installed with Visual Studio, this fails.

Is there a property set in the build or a way to get a binlog from that?

**Comment by zivkan** [comment]
`dotnet list package` doesn't build the project, but it does use MSBuild APIs to read project properties, which means that MSBuild needs to be able to load (["evaluate"](https://learn.microsoft.com/en-us/visualstudio/msbuild/build-process-overview?view=vs-2022#evaluation-phase)) the project. Since `dotnet list package` is part of the .NET SDK, msbuild uses the targets/props from the .NET SDK, and therefore only officially supports projects that the .NET SDK supports.

Unofficially, I think I once saw someone set some MSBuild environment variables that overrides what directory MSBuild uses to load props/targets from. This won't always work, because if a target runs, and the target runs a task that only works on .NET Framework, then it can't work on the .NETCoreApp msbuild. That probably won't be a problem for `dotnet list package` though, so as long as MSBuild evaluation works `dotnet list package` will probably work. It just needs the project to have been restored via msbuild first. Having a quick look at [msbuild's reserved and well-known properties](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-reserved-and-well-known-properties?view=vs-2022), my guess is that setting `MSBuildExtensionsPath` to VS's MSBuild directory might be it. Zero guarantees that it'll work though.

**Reply by paulomorgado** [reply]
Changing `MSBuildExtensionsPath` might cause other issues.

`dotnet list package` could inject a property, like Visual Studio injects `DesignTimeBuild`.

But it's probably not worth the effort.
